sagemaker/launch/launch_realtime_endpoint.py

```diff
@@ -0,0 +1,269 @@
+                          check_tarfile_s3_object,
+                          parse_s3_url,
+                          upload_data_to_s3,
+                          check_name_format)
+
+
+# TODO: When adding new realtime inference tasks, modify this list
+SUPPORTED_REALTIME_INFER_TASKS = ['node_classification']
+
+# Constants for SageMaker endpoints
+ENTRY_FOLDER_NAME = 'realtime_entry_points'
+ENTRY_FILE_NAMES = [
+    'node_classification_entry.py',
+    'node_regression_entry.py',
+    'edge_classification_entry.py',
+    'edge_regression_entry.py']
+DEFAULT_GS_MODLE_FILE_NAME = 'model.pt'
+
+
+def run_job(input_args):
+    """ The procedure of deploying a SageMaker real-time inference endpoint
+    
+    Following SageMaker's document for deploying model for real-time inference in
+    https://docs.aws.amazon.com/sagemaker/latest/dg/realtime-endpoints-deploy-models.html,
+    the general steps of the deployment include:
+    1. prepare model artifacts;
+    2. create a deployable model;
+    3. create an endpoint configuration based on the deployable model;
+    4. create an endpoint based on the endpoint configuration.
+
+    This job follows these steps.
+    """
+    # ================= prepare model artifacts ================= #
+    # prepare sessions
+    b3_session = boto3.Session(region_name=input_args.region)
+    sm_session = sm.Session(boto_session=b3_session)
+    
+    # If users provide the S3 tar file path already, directly use it for model creation
+    if input_args.model_tarfile_s3:
+        entrypoint_file_name = input_args.entrypoint_file_name
+
+        model_url_s3 = input_args.model_tarfile_s3
+        assert check_tarfile_s3_object(args.model_tarfile_s3) == True, \
+            (f'Not find a tar file in the given S3 URL: {args.model_tarfile_s3}...')
+
+        # model_name is either directly given by users, or had been assigned by the argument
+        # validation function
+        model_name = input_args.model_name
+    else:
+        # prepare task dependent variables
+        entry_point_dir = os.path.join(os.path.dirname(__file__), ENTRY_FOLDER_NAME)
+
+        # TODO: When adding new realtime inference tasks, add new elif here to support them
+        if input_args.infer_task_type == SUPPORTED_REALTIME_INFER_TASKS[0]: # node_classification
+            entrypoint_file_name = list(ENTRY_FILE_NAMES)[0]
+            path_to_entry = os.path.join(entry_point_dir, entrypoint_file_name)
+        else:
+            raise NotImplementedError(f'The given real-time inference task \
+                                        {input_args.infer_task_type} is not supported.')
+
+        path_to_model = os.path.join(input_args.restore_model_path, DEFAULT_GS_MODLE_FILE_NAME)
+        path_to_yaml = input_args.model_yaml_config_file
+        path_to_json = input_args.graph_json_config_file
+        model_name = input_args.model_name
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
```

Comment:
Can we use the current dir (`pwd`) as the root directory for storing the temporal tar file? User may not have the permission to create files in /tmp/xxx

Reply:
Yes, modified to use the current dir and a temporary folder for packing. After uploading, remove the temporary folder.